machine/v1beta1/types_vsphereprovider.go

```diff
@@ -172,6 +181,17 @@ type NetworkDeviceSpec struct {
 	AddressesFromPools []AddressesFromPool `json:"addressesFromPools,omitempty"`
 }
 
+// VSphereDisk describes additional disks for vSphere.
+type VSphereDisk struct {
```

Comment:
While markers don't actually apply to these types, I'd prefer if we could add them so that we can show what limits we are imposing via the webhooks

We should have a maxlength on the name, and character set validation if at all possible

On the size, a minimum and maximum should be imposed

Reply:
Sounds good.  I'll make those updates.

Reply:
I updated the name to have a validation and max length of 80.  I also set the max disk size to 62TB (57742 GiB) since that is the max allowed in vSphere.